main.py

```diff
@@ -25,6 +31,129 @@
+        return
+    if db.push_resume(ctx.message.author.id, args[0]).is_success:
+        await bot.send_message(
+            ctx.message.channel,
+            "successfully added you to the resume queue.\n" +
+            "PSA: please anonymize your resumes. You can replace your resume and keep your spot with !resume replace.")
+    else:
+        await bot.send_message(ctx.message.channel, "you already have a resume in the queue. Try !resume replace [resume].")
+    
+@resume.group(name='delete', aliases=['delete'], pass_context=True)
+async def remove(ctx, *args):
+    global db
+    if len(args) != 0:
+        await bot.send_message(ctx.message.channel, error_message)
+        return
+    if db.delete_resume(ctx.message.author.id).is_success:
+        await bot.send_message(ctx.message.channel, "successfully deleted your resume.")
+    else:
+        await bot.send_message(ctx.message.channel, "you don't have a resume in the queue.")
+
+@resume.group(name='replace', pass_context=True)
+async def replace(ctx, *args):
+    global db
+    if len(args) != 1:
+        await bot.send_message(ctx.message.channel, error_message)
+        return
+    if db.replace_resume(ctx.message.author.id, args[0]).is_success:
+        await bot.send_message(ctx.message.channel, "successfully replaced your resume.")
+    else:
+        await bot.send_message(ctx.message.channel, "you don't have a resume in the queue.")
+
+@resume.group(name='poll', aliases=['pop'], pass_context=True)
+async def poll(ctx, *args):
+    global db
+    if len(args) != 0:
+        await bot.send_message(ctx.message.channel, error_message)
+        return
+    query_result = db.pop_resume()
+    if query_result.is_success:
+        (user_id, resume) = query_result.data
+        user = await bot.get_user_info(user_id)
+        await bot.send_message(ctx.message.channel, f"resume by {user}: <{resume}>")
+    else:
+        await bot.send_message(ctx.message.channel, "there are no resumes currently in the queue.")
+
+@resume.group(name='peek', pass_context=True)
+async def peek(ctx, *args):
+    global db
+    if len(args) != 0:
+        await bot.send_message(ctx.message.channel, error_message)
+        return
+    result = db.show_resumes(1)
+    if result.is_success:
+        msg_content = ["resumes currently in the queue:"]
+        for (user_id, resume) in result.data:
+            user = await bot.get_user_info(user_id)
+            msg_content.append(f"{user}: <{resume}>")
+        await bot.send_message(ctx.message.author, "\n".join(msg_content))
+    else:
+        await bot.send_message(ctx.message.channel, "there are no resumes currently in the queue.")
+
+@resume.group(name='show', pass_context=True)
+async def show(ctx, *args):
+    global db
+    if len(args) != 1 and len(args) != 0:
+        await bot.send_message(ctx.message.channel, error_message)
+        return
+    if len(args) == 0:
+        n = 3
+    elif args[0] == "all":
+        n = 999
```

Comment:
Why have a default max?  If anything it might be better to just say here's the first 10 results, send !next to see the next 10 (or something like this).

Reply:
It's leftover logic from the Javascript version. We could probably just add a ``show_all_resumes`` method to Database now.

Reply:
With the amount parameter now being optional, we no longer need the default max